includes/abstracts/llms-abstract-query.php

```diff
@@ -0,0 +1,589 @@
+
+		// If we have results and found rounds isn't disabled.
+		if ( $this->number_results && ! $this->get( 'no_found_rows' ) ) {
+			$this->found_results = $this->found_results();
+			$this->max_pages     = absint( ceil( $this->found_results / $this->get( 'per_page' ) ) );
+		}
+
+	}
+
+	/**
+	 * Determines the default arguments for the query.
+	 *
+	 * Extending classes can override or extend this method to customize the default query
+	 * arguments for the query.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	protected function default_arguments() {
+
+		return array(
+			'page'             => 1,
+			'per_page'         => 10,
+			'search'           => '',
+			'sort'             => array(),
+			'suppress_filters' => false,
+			'no_found_rows'    => false,
+		);
+
+	}
+
+	/**
+	 * Determines the total number of found results for the given query and returns it.
+	 *
+	 * @since [version]
+	 *
+	 * @return int
+	 */
+	abstract protected function found_results();
+
+	/**
+	 * Retrieve a query variable with an optional fallback value when the value is not set.
+	 *
+	 * @since 3.8.0
+	 * @since [version] Moved from `LLMS_Database_Query` and updated to use the null coalesce operator.
+	 *
+	 * @param string $key     Variable key.
+	 * @param mixed  $default Default value.
+	 * @return mixed
+	 */
+	public function get( $key, $default = '' ) {
+		return $this->query_vars[ $key ] ?? $default;
+	}
+
+	/**
+	 * Get the final combined arguments used to generate the query.
+	 *
+	 * Retrieves the value of the protected `$arguments` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_arguments() {
+		return $this->arguments;
+	}
+
+	/**
+	 * Get the default arguments used for the query.
+	 *
+	 * Retrieves the value of the protected `$arguments_default` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_arguments_default() {
+		return $this->arguments_default;
+	}
+
+	/**
+	 * Get the original, uncleaned, arguments submitted to the query.
+	 *
+	 * Retrieves the value of the protected `$arguments_original` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_arguments_original() {
+		return $this->arguments_original;
+	}
+
+	/**
+	 * Get the
+	 *
+	 * Retrieves the value of the protected `$query` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return mixed
+	 */
+	public function get_query() {
+		return $this->query;
+	}
+
+	/**
+	 * Retrieve a list of fields that are allowed to be used for result sorting.
+	 *
+	 * @since [version]
+	 *
+	 * @return string[]
+	 */
+	protected function get_allowed_sort_fields() {
+
+		$allowed_fields = $this->allowed_sort_fields;
+
+		if ( $this->get( 'suppress_filters' ) ) {
+			return $allowed_fields;
+		}
+
+		/**
+		 * Filters the allowed sort fields.
+		 *
+		 * The dynamic portion of this hook, `$this->type`, refers to the award type
+		 * for the query, either "achievements" or "certificates".
```

Comment:
This is not for award types only anymore.
```suggestion
		 * The dynamic part of the filter `$this->id` identifies the extending query.
```